pkg/apis/pipeline/v1alpha1/pipelinerun_types.go

```diff
@@ -72,9 +72,9 @@ type PipelineTriggerRef struct {
 
 // PipelineRunStatus defines the observed state of PipelineRun
 type PipelineRunStatus struct {
+	Conditions duckv1alpha1.Conditions `json:"conditions"`
 	//+optional
-	TaskRuns []PipelineTaskRun `json:"taskRuns,omitempty"`
-	Conditions      duckv1alpha1.Conditions   `json:"conditions"`
+	TaskRuns map[string]TaskRunStatus `json:"taskRuns,omitempty"`
```

Comment:
Can you add a comment about what the key is here? Is it the task name, or the task run ID?

Can this just be a `[]TaskRunStatus` where indexes match up with the indexes in the `PipelineRun`'s referenced `Pipeline`'s `Task`s?

Reply:
the key is currently the taskrun name.
The taskRuns in some cases can run in parallel so the order might be different than how the pipeline is defined, I thought it would be easier to process and eventually visualize with a map to let you easily get the required task

Reply:
just a bump here to @ImJasonH 's request to add a comment explaining what the keys in this map are